packages/maskbook/src/plugins/ITO/UI/ClaimDialog.tsx

```diff
@@ -206,8 +214,7 @@ export function ClaimDialog(props: ClaimDialogProps) {
         if (claimAmount.isEqualTo(0)) return t('plugin_ito_error_enter_amount')
         if (claimAmount.isGreaterThan(new BigNumber(tokenBalance)))
             return t('plugin_ito_error_balance', { symbol: claimToken.symbol })
-        if (claimAmount.dividedBy(ratio).isGreaterThan(maxSwapAmount))
```

Comment:
Compare token amount state with state directly rather than with the calculating result which is not precise.

E.g. after click the max button, or input the max number or slide to the max

<img src="https://user-images.githubusercontent.com/63733714/107137634-643a3f00-6949-11eb-944a-8423bf56d32e.png" width=400 />


```ts
console.log(tokenAmount.toFixed()) // 601424451774961270 ✅
console.log(maxSwapAmount.toFixed()) // 601424451774961270 ✅
console.log(claimAmount) // 79388028 ✅
console.log(claimAmount.dividedBy(ratio).toFixed()) // 601424454545454545.45454545454545454545 ❌
``` 

So, why `claimAmount.dividedBy(ratio) !== tokenAmount` ? 
[ClaimDialog.tsx/R137](https://github.com/DimensionDev/Maskbook/pull/2360/files#diff-30e825f79dbac68533e6dfe3b6c19c4baf2fba760b3631a5f83d9404d0da491eR137)

```ts
const maxAmount = useMemo(() => BigNumber.min(maxSwapAmount.multipliedBy(ratio).dp(0), tokenBalance).toFixed(), 
```

We lose the precision by call `.dp(0)`, so why call `dp(0)` here?

[TokenAmountPanel.tsx#L145](https://github.com/DimensionDev/Maskbook/blob/master/packages/maskbook/src/web3/UI/TokenAmountPanel.tsx#L145): 

```
formatBalance(new BigNumber(maxAmount ?? balance), token.decimals)
```

Because we need to pass maxAmount to formatBalance which only handles integer.

I think we may should refactor props of `<TokenAmountPanel />` from, because set a value which indicates BigNumber with string type by using `formatBalance()` or `.toFixed()` could lead to precision losing.

```ts
amount: string
maxAmount?: string
balance: string
```

to: 

```ts
amount: BigNumber
maxAmount?: BigNumber
balance: BigNumber
```